substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/serialize/SerializationRegistry.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2019, 2020, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2020 Alibaba Group Holding Limited. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.core.jdk.serialize;
+
+import com.oracle.svm.core.jdk.Package_jdk_internal_reflect;
+import org.graalvm.nativeimage.Platform;
+import org.graalvm.nativeimage.Platforms;
+import org.graalvm.nativeimage.hosted.Feature.FeatureAccess;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public interface SerializationRegistry {
+    @Platforms(Platform.HOSTED_ONLY.class)
+    Class<?> addSerializationConstructorAccessorClass(Class<?> serializationTargetClass, Long checksum);
+
+    @Platforms(Platform.HOSTED_ONLY.class)
+    void reportErrors(FeatureAccess access);
+
+    Object getSerializationConstructorAccessorClass(Class<?> serializationTargetClass, String targetConstructorClass);
+
+    @Platforms(Platform.HOSTED_ONLY.class)
+    static Object createSerializationConstructorAccessorClass(Class<?> serializationTargetClass, Class<?>[] parameterTypes, Class<?>[] checkedExceptions,
+                    int modifiers, Class<?> targetConstructorClass) {
+        try {
+            Class<?> generatorClass = Class.forName(Package_jdk_internal_reflect.getQualifiedName() + ".MethodAccessorGenerator");
+            Constructor<?> c = generatorClass.getDeclaredConstructor();
+            c.setAccessible(true);
+            Object generator = c.newInstance();
```

Comment:
Please use `ReflectionUtil.newInstance`, which also takes care of the module system, and `ReflectionUtil.lookupMethod` for below.

Reply:
fixed